api/envoy/config/core/v3/extension.proto

```diff
@@ -28,3 +30,30 @@ message TypedExtensionConfig {
   // <config_overview_extension_configuration>` for further details.
   google.protobuf.Any typed_config = 2 [(validate.rules).any = {required: true}];
 }
+
+// Configuration source specifier for a late-bound
+// extension configuration. The parent resource is warmed until all the initial
+// extension configurations are received, unless the flag to apply the default
+// configuration is set. Subsequent extension updates are atomic on a per-worker
+// basis, and apply to new streams while the active streams continue using
+// the older extension configurations. If the initial delivery of the extension
+// configuration fails, due to a timeout for example, the optional default
+// configuration is applied. Without a default configuration, the extension is
+// disabled, until an extension configuration is received.
+message ExtensionConfigSource {
+  ConfigSource config_source = 1;
+
+  // Optional default configuration to use as the initial configuration if
+  // there is a failure to receive the initial extension configuration or if
+  // `apply_default_config_without_warming` flag is set.
+  google.protobuf.Any default_config = 2;
+
+  // Use the default config as the initial configuration without warming and
+  // waiting for the first discovery response. Requires the default configuration
+  // to be supplied.
+  bool apply_default_config_without_warming = 3;
+
+  // Optional constraint on the type URL. Extension configuration updates are rejected
```

Comment:
Do we need to support optional? I.e. is there a use case where we would not know ahead of time the type of an extension with a given name.

Reply:
Yeah, upgrade. For that, we might need a set of type URLs.

Reply:
Configuration or extension implementation upgrade? Envoy built-in or proprietary config?

Reply:
Configuration update. It seems that having a single type URL is too restrictive. I am fine with making it a set and mandatory. Would that work?

Reply:
Note that for Wasm extension, type URL doesn't really supply enough information to constrain it. We might need some per-extension admin controls.

Reply:
Yeah, a mandatory set is fine. For Wasm, I think all we can validate here is that we are instantiating some Wasm, which is fine. There aren't Wasm specific validations to perform.

Reply:
Ack.